src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs

```diff
@@ -540,6 +556,12 @@ private async Task EnsureModulesLoaded()
             }
         }
 
+        private async Task UpdateTargetFeatures()
+        {
+            TargetFeatures = await MICommandFactory.GetTargetFeatures();
+            // TODO: notify OpenDebugAD7 to run AD7DebugSession.UpdateCapabilities
```

Comment:
Wouldn't know how. Just an idea to make things more asynchronous rather than checking reversibility on every stop event.
But yeah, no major further development planned. Just want to get these old PRs merged.

Reply:
Got it. In that case I would probably suggest changing the comment to something like:

`// NOTE: Currently OpenDebugAD7 polls to decide if we can reverse execute. It might make sense to push an event. If we ever create such an event, probably makes sense to use it here also.`

Reply:
Ok. Btw the events are RecordStartedEvent and RecordStoppedEvent for `gdb`. And we check it after starting up to support `rr`.